internal/operators/ocs/validations.go

```diff
@@ -128,40 +123,55 @@ func (o *ocsValidator) ValidateRequirements(cluster *models.Cluster) (api.Valida
 	return api.Failure, status
 }
 
-func (o *ocsValidator) nodeResources(host *models.Host, cpuCount *int64, totalRAM *int64, diskCount *int64, hostsWithDisks *int64, insufficientHosts *[]string) (bool, error) {
+func (o *operator) checkInventoryAndNodeRequirements(r resources) (api.ValidationStatus, error) {
+	inventoryMissing, err := o.nodeResources(r)
+	if err != nil {
+		o.log.Error("Error occurred while calculating Node requirements ", err)
+		return api.Failure, errors.New("Error occurred while calculating Node requirements ")
+	} else if inventoryMissing {
+		return api.Pending, errors.New("Missing Inventory in some of the hosts ")
+	}
+	return "", nil
+}
+
+func (o *operator) nodeResources(r resources) (bool, error) {
 	var inventory models.Inventory
 	// if inventory is empty
-	if host.Inventory == "" {
-		o.log.Info("Empty Inventory of host with hostID ", *host.ID)
+	if r.host.Inventory == "" {
+		o.log.Info("Empty Inventory of host with hostID ", *r.host.ID)
 		return true, nil // to indicate that inventory is empty and the ValidationStatus must be Pending
-	} else if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
-		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
+	} else if err := json.Unmarshal([]byte(r.host.Inventory), &inventory); err != nil {
+		o.log.Errorf("Failed to get inventory from host with id %s", r.host.ID)
 		return false, err
 	}
 
 	disks := getValidDiskCount(inventory.Disks)
 
 	if disks > 1 { // OCS must use the non-boot disks
-		requiredDiskCPU := int64(disks-1) * o.OCSRequiredDiskCPUCount
-		requiredDiskRAM := int64(disks-1) * o.OCSRequiredDiskRAMGB
-
-		if inventory.CPU.Count < requiredDiskCPU || inventory.Memory.UsableBytes < gbToBytes(requiredDiskRAM) {
-			status := fmt.Sprint("Insufficient resources on host with host ID ", *host.ID, " to deploy OCS. The hosts has ", disks, " disks that require ", requiredDiskCPU, " CPUs, ", requiredDiskRAM, " RAMGB")
-			*insufficientHosts = append(*insufficientHosts, status)
-			o.log.Info(status)
-		} else {
-			*cpuCount += (inventory.CPU.Count - requiredDiskCPU)                     // cpus excluding the cpus required for disks
-			*totalRAM += (inventory.Memory.UsableBytes - gbToBytes(requiredDiskRAM)) // ram excluding the ram required for disks
-			*diskCount += (int64)(disks - 1)                                         // not counting the boot disk
-			*hostsWithDisks++
-		}
+		o.countResources(&inventory, r, int64(disks))
 	} else {
-		*cpuCount += inventory.CPU.Count
-		*totalRAM += inventory.Memory.UsableBytes
+		*r.cpuCount += inventory.CPU.Count
+		*r.totalRAM += inventory.Memory.UsableBytes
 	}
 	return false, nil
 }
 
+func (o *operator) countResources(inventory *models.Inventory, r resources, disks int64) {
```

Comment:
rename to `countResourcesExcludingDiskResources` and add a comment that `OCS requires 2 cpus, 5GB ram` per disk